samples/csharp/getting-started/Ranking_Web/README.md

```diff
@@ -0,0 +1,155 @@
+
+## ML Task - Ranking
+As previously mentioned, this sample uses the LightGBM LambdaRank algorithm which is applied using a supervised learning technique known as [**Learning to Rank**](https://en.wikipedia.org/wiki/Learning_to_rank).  This technique requires that train\test datasets contain groups of data instances that are each labeled with their relevance score (e.g. relevance judgment label).  The label is a numerical\ordinal value, such as {0, 1, 2, 3, 4}.  The process for labeling these data instances with their relevance scores can be done manually by subject matter experts.  Or, the labels can be determined using other metrics, such as the number of clicks on a given search result. 
+
+It is expected that the dataset will have many more "Bad" relevance scores than "Perfect".  This helps to avoid converting a ranked list directly into equally sized bins of {0, 1, 2, 3, 4}.  The relevance scores are also reused so that you will have many items **per group** that are labeled 0, which means the result is "Bad".  And, only one or a few labeled 4, which means that the result is "Perfect". 
+
+Once the train\test datasets are labeled with relevance scores, the model (e.g. ranker) can then be trained and tested using this data.  Through the model training process, the ranker learns how to score each data instance within a group based on their label value.  The resulting score of an individual data instance by itself isn't important -- instead, the scores should be compared against one another to determine the relative ordering of a group's data instances.  The higher the score a data instance has, the more relevant and more highly ranked it is within its group.      
+
+## Solution
+Since this sample's dataset already is already labeled with relevance scores, we can immediately start with training the model.  In cases where you start with a dataset that isn't labeled, you will need to go through this process first by having subject matter experts provide relevance scores or by using some other metrics to determine relevance.
+
+This sample performs the following high-level steps to rank the search engine results:
+1. The model is **trained** using the train dataset with LightGBM LambdaRank.  
+2. The model is **tested** using the test dataset.  This results in a **prediction** that includes a **score** for each search engine result.  The score is used to determine the ranking relative to other results within the same query (e.g. group).  The predictions are then **evaluated** by examining metrics; specifically the [Normalized Discounted Cumulative Gain](https://en.wikipedia.org/wiki/Discounted_cumulative_gain)(NDCG).
+3. The final step is to **consume** the model to perform ranking predictions for new incoming searches.
```

Comment:
We would be better describing the full data science process of training a model. Mainly we (our sample code) should push the need of refitting on all available data when creating the final model, and that train/test or CV splits are only for getting metrics.

Quoted from: https://github.com/dotnet/machinelearning-samples/pull/533#discussion_r298744919
> ...let users experiment on the dataset by iterating using the scores on the validation set, then testing their final solution, exactly once, on the test dataset.
> 
> Generally the DS pattern for ML should look like:
> 1. Train on training set, get metrics from validation set
> 2. (iterate many times until happy with pipeline -- goto 1)
> 3. Train on found pipeline on combined train+validate set, get metrics on test set (exactly once) -- this is your final metrics
> 4. Retrain  pipeline on all data train+validate+test set. Send this newly created model to production.
> 
> The final estimate of how well your model will do in production is the metrics from step (3). The final model for production, trained on all available data, is trained at step (4).

Reply:
@justinormont  - I modified the ReadMe and the code to follow the steps that you outlined above.  This resulted in quite a few changes. Can you take another look to make sure that I understood everything correctly?  Thanks!

Reply:
#resolved